setup.py

```diff
@@ -0,0 +1,80 @@
+"""
+Setup file
+"""
+
+# To use a consistent encoding
+import codecs
+import os
+
+# Always prefer setuptools over distutils
+from setuptools import setup, find_packages
+
+
+def get_long_description():
+    """
+    Get the long description from the README file
+    """
+    here = os.path.abspath(os.path.dirname(__file__))
+
+    with codecs.open(os.path.join(here, 'README.md'), encoding='utf-8') as readme:
+        return readme.read()
+
+
+def find_package_data(package, data_dir):
+    """
+    Find all the package data
+    """
+    package_data = []
+    oldcwd = os.getcwd()
+    os.chdir(package)
+    for dirpath, _, filenames in os.walk(data_dir):
+        files = [os.path.join(dirpath, filename) for filename in filenames]
+        package_data.extend(files)
+    os.chdir(oldcwd)
+    return package_data
+
+
+setup(
+    name='cli',
+    version='0.0.1',
+    description='packaging wrapper using ansible',
+    long_description=get_long_description(),
+    long_description_content_type='text/markdown',
+    url='https://github.com/theforeman/foreman-quadlet',
+    author='The Foreman Project',
+    author_email='foreman-dev@googlegroups.com',
+    zip_safe=False,
+
+    classifiers=[
+        'Development Status :: 3 - Alpha',
+        'Intended Audience :: Developers',
+        'Topic :: Software Development :: Build Tools',
+        'License :: OSI Approved :: GNU General Public License v2 (GPLv2)',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.9',
+        'Programming Language :: Python :: 3.10',
+        'Programming Language :: Python :: 3.11',
+        'Programming Language :: Python :: 3.12',
+        'Programming Language :: Python :: 3.13',
+    ],
+
+    keywords='foreman',
+
+    packages=find_packages(exclude=['contrib', 'docs', 'tests']),
+
+    install_requires=['obsah >= 1.0.0'],
+
+    extras_require={
+        'argcomplete': ['argcomplete'],
+    },
+
+    package_data={
+        'cli': find_package_data('cli'),
+    },
+
+    entry_points={
+        'console_scripts': [
+            'cli=cli:main',
```

Comment:
This creates a command `cli` but don't you want to name this `rop`? Then it would also remove the need for a wrapper script.
```suggestion
            'rop=cli:main',
```

Reply:
Right, but I wanted to get rid of the wrapper script you have now. The `console_scripts` entry point will work nicely once we start packaging things up in RPMs etc.

Reply:
How do I call it without a wrapper script from git checkout?

Reply:
The workflow becomes
```console
git clone https://github.com/theforeman/foreman-quadlet
cd foreman-quadlet
# create a virtualenv and activate it
pip install -e .
pip install -r requirements.txt # we should merge these 2 together
```

Now in your virtualenv you have the wrapper script available just like it would be in production, calling the correct Python.

uv is also a new tool that makes packaging single scripts easier: https://docs.astral.sh/uv/guides/scripts/#running-a-script-with-dependencies. Perhaps we should investigate that approach too.

Reply:
My view was the native python packaging can happen after we've pioneered the start of the CLI as that can be a re-factoring on top of a working model. I'd like to explore the question of:

>  "Related to this is discussions about if this helps us provide an equivalent to the [operations collection](https://github.com/theforeman/foreman-operations-collection) more natively."

As I wonder about shipping the CLI bits as Python and the Ansible bits as a collection. Is it possible for it to work that way? can Obsah consume some Ansible bits as a collection? I'd assume so since we use modules that come from collections in Obsah.

Reply:
The way I view it we'll still have some CLI bits (likely as Python), but we can ship the actual Ansible bits separate. That effectively ends up in a similar situation like foreman-installer with separate Puppet modules.

Big question is: do we vendor the collections in packaging like we do with Puppet or ship them separate.

Reply:
Also, having some artifact we can install in a production like setup can make testing with tmt easier. That was my angle.